app/utilities/drawing/utils.py

```diff
@@ -0,0 +1,125 @@
+import sys
+import PIL
+from PIL import Image, ImageDraw, ImageOps, ImageFont, ImageEnhance, ImageFilter
+
+from . import constants
+
+
+def drawBadgeBackgroundOnBackground(offset, background):
+    black_region = PIL.Image.new(mode="RGBA", size=constants.badge_size, color=(255, 255, 255, 25))
+    background.paste(black_region, (offset[0], offset[1]), create_rounded_rectangle_mask(black_region, 8))
+    return background
+
+def drawBadgeImageOnBackground(offset, background, badge_name):
+    badge_path = constants.badges_map[badge_name]
+    black_region = Image.open(badge_path)
+    black_region = black_region.resize(constants.badge_size, Image.Resampling.LANCZOS)
+    background.paste(black_region, (offset[0], offset[1]), create_rounded_rectangle_mask(black_region, 8))
+    return background
+
+def createSemiTransparentBlackRegionOnBackground(size, offset, background):
+    cropped_img = background.crop((
+        offset[0],
+        offset[1],
+        size[0]+offset[0],
+        size[1]+offset[1]
+    ))
+    alpha_bg = PIL.Image.new(mode="RGBA", size=cropped_img.size, color=(18, 17, 21, 127))
+    black_region = Image.alpha_composite(cropped_img, alpha_bg)
+    background.paste(black_region, (offset[0], offset[1]), create_rounded_rectangle_mask(black_region, 12))
+    return background
+
+def applyAlphaWithMask(image, mask_path):
+    mask = Image.open(mask_path).convert('L')
+    output = ImageOps.fit(image, mask.size, centering=(0.5, 0.5))
+    output.putalpha(mask)
+    return output
+
+def drawText(text, size, location, weight, draw, color=(255,255,255)):
+    font_path = constants.default_font
+    font = ImageFont.truetype(font_path, size)
+    font.set_variation_by_name(weight)
+    draw.text(location, text, color, font=font)
+
+def create_rounded_rectangle_mask(rectangle, radius):
+    solid_fill =  (50,50,50,255)
+    # create mask image. all pixels set to translucent
+    i = Image.new("RGBA",rectangle.size,(0,0,0,0))
```

Comment:
```suggestion
    i = Image.new("RGBA", rectangle.size, (0,0,0,0))
```

Maybe also rename `i` to `image` or `img`?